### PR TITLE
chore: Allow logging without inheritance

### PR DESCRIPTION
**Pull request by sss04**
## Related Issues/PRs

#xxx

## What changes are proposed in this pull request?

Add log_verb_static - a static function that doesn't depend on the SynapseMLLogger class to log, instead opting to use self.logger in the class it's used in. 

Yet to add functionality to log more info. 

## How is this patch tested?

- [ ] I have written tests (not required for typo or doc fix) and confirmed the proposed feature/bug-fix/change works.

## Does this PR change any dependencies?

- [ ] No. You can skip this section.
- [ ] Yes. Make sure the dependencies are resolved correctly, and list changes here.

## Does this PR add a new feature? If so, have you added samples on website?

- [ ] No. You can skip this section.
- [ ] Yes. Make sure you have added samples following below steps.

1. Find the corresponding markdown file for your new feature in `website/docs/documentation` folder.
   Make sure you choose the correct class `estimators/transformers` and namespace.
2. Follow the pattern in markdown file and add another section for your new API, including pyspark, scala (and .NET potentially) samples.
3. Make sure the `DocTable` points to correct API link.
4. Navigate to website folder, and run `yarn run start` to make sure the website renders correctly.
5. Don't forget to add `<!--pytest-codeblocks:cont-->` before each python code blocks to enable auto-tests for python samples.
6. Make sure the `WebsiteSamplesTests` job pass in the pipeline.

### CODE REVIEW

**Review by mhamilton723 — DISMISSED** [review]
Lovely work!

File: core/src/main/python/synapse/ml/core/logging/SynapseMLLogger.py [hunk 1]
```diff
@@ -213,6 +218,58 @@ def log_decorator_wrapper(self, *args, **kwargs):
 
         return get_wrapper
 
+    @staticmethod
+    def log_verb_static(method_name: Optional[str] = None,  feature_name: Optional[str] = None, custom_log_function = None):
+        def get_wrapper(func):
+            @functools.wraps(func)
+            def log_decorator_wrapper(self, *args, **kwargs):
+                # Validate that self.logger is set
+                if not hasattr(self, "logger"):
```
Comment (TomFinley) [inline core/src/main/python/synapse/ml/core/logging/SynapseMLLogger.py:227]: If it has `logger` as an atttribute, this becomes part of the public API. I suggest something like `_logger`. Or even make it configurable like the method name is configurable.

**Review by TomFinley — COMMENTED** [review]
Just a note about not putting the `logger` instance in the public API.

File: core/src/main/python/synapse/ml/core/logging/SynapseMLLogger.py (hunk 1 above)
Comment (sss04) [reply core/src/main/python/synapse/ml/core/logging/SynapseMLLogger.py:227]: I can make the change to `_logger` for now

**Review by mhamilton723 — APPROVED** [review]

**Pull request merged**